app/components/more-details.jsx

```diff
@@ -0,0 +1,146 @@
+  const options = schema.enum || []
+  const label = schema.title || uischema.label
+
+  let type
+  if (schema.format === 'date') {
+    type = 'date'
+  } else if (schema.type === 'integer' || schema.type === 'number') {
+    type = 'number'
+  } else if (schema.type === 'boolean') {
+    type = 'checkbox'
+  } else if (options.length > 0) {
+    type = 'select'
+  } else {
+    type = 'text'
+  }
+
+  return (
+    <div>
+      <label>{label}</label>
+      {type === 'select' ? (
+        <select value={data || ''} onChange={i => handleChange(path, i.target.value)} className={classes.formInput}>
+          <option value="" disabled>
+            Choose one
+          </option>
+          {options.map(option => (
+            <option key={option} value={option}>
+              {option}
+            </option>
+          ))}
+        </select>
+      ) : (
+        <input
+          type={type}
+          checked={type === 'checkbox' ? data : undefined}
+          value={type === 'checkbox' ? undefined : data || ''}
+          onChange={i => handleChange(path, type === 'checkbox' ? i.target.checked : i.target.value)}
+          className={classes.formInput}
+        />
+      )}
+    </div>
+  )
+})
+
+const customRenderers = [...vanillaRenderers, { tester: rankWith(3, isControl), renderer: CustomInputRenderer }]
+
+const MoreDetails = props => {
+  const { className = '', schema = {}, uischema = {}, details = {}, onDone = () => {}, ...otherProps } = props
+  const { title } = otherProps
+  const [data, setData] = useState(details)
+  const [isValid, setIsValid] = useState(false)
+  const classes = useStyles(props)
+
+  const handleOnClick = () => {
+    if (isValid) {
+      onDone({ valid: true, value: data })
+    }
+  }
+
+  const handleIsValid = () => {
+    const requiredData = schema.properties || {}
+    return Object.keys(requiredData).every(i => (requiredData[i].enum ? data[i] !== undefined && data[i] !== '' : true))
+  }
+
+  useEffect(() => {
+    setIsValid(handleIsValid())
+  }, [data, schema])
+
+  return (
+    <div className={cx(classes.formContainer, className)}>
+      {title && <p className={classes.formTitle}>{title}</p>}
+      <div className={classes.jsonFormContainer}>
+        <JsonForms
+          schema={schema}
+          uischema={uischema}
+          data={data}
+          renderers={customRenderers.map(renderer => ({
+            ...renderer,
+            renderer:
+              renderer.renderer === CustomInputRenderer
+                ? props => <CustomInputRenderer {...props} classes={classes} />
+                : renderer.renderer,
+          }))}
+          cells={vanillaCells}
+          onChange={({ data, _errors }) => setData(data)}
+        />
+        <PrimaryButton
+          primary
+          tile={'Submit form'}
+          className={classes.submitButton}
+          onClick={handleOnClick}
```

Comment:
Do you mean each story should have onDoneResults showing? Because I only added it to the story marked "Create a stories that tests users inputting data and onDone being called" in the issue, since it was specifically stated there. I can add it to the rest as well.